### ISSUE TITLE
Field's label can't be set locally to inform assistive technology users what to input (1.3.1)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux-lookup#95]
### Expected behavior

- Input field has a label programatically associated to be provided to assistive technology users about what to input. 
- Teams can associate the input to their context-specific label or provide a label.

In example below, the input label should be "Country of residence"
![image](https://user-images.githubusercontent.com/15878597/82561149-dd3a0200-9b40-11ea-82df-d76f343e1e1c.png)

### Actual behavior
The input has an aria-label="Search for a country". Aria-label is at the top of the hierarchy in the algorithm browsers use to determine an input's label. Users will always get this as the label and not whatever the label needs to be given what's being asked for (e.g., Country of residence, Country of birth, Going to, Coming from).

### Impact
Assistive technology users won't be communicated correct input field label causing confusion and/or incorrect input.
